cms/static/js/factories/login.js

```diff
@@ -22,15 +22,19 @@ define(['jquery.cookie', 'utility'], function() {
 
         $('form#login_form').submit(function(event) {
             event.preventDefault();
+            var submit_button = $('#submit'),
+                deferred = new $.Deferred(),
+                promise = deferred.promise();
+            ViewUtils.disableElementWhileRunning(submit_button, function() { return promise; });
```

Comment:
@andy-armstrong i didnt get the point here, can you explain more what we want to do here ?